tensorflow/c/kernels.cc

```diff
@@ -455,6 +455,20 @@ TF_DataType TF_ExpectedOutputDataType(TF_OpKernelContext* ctx, int i) {
   return static_cast<TF_DataType>(cc_ctx->expected_output_dtype(i));
 }
 
+bool TF_IsHostMemoryInput(TF_OpKernelContext* ctx, int i) {
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(ctx);
+  CHECK_GE(i, 0);
```

Comment:
I'm a little wary of a CHECK failure here - could we return a TF_Status* instead? Here and below